### PR TITLE
[DEV-1205] Port CI to GHA

### PR DESCRIPTION
**Pull request by vmarkovtsev**

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        key: ${{ env.GO_CACHE_PREFIX }}${{ hashFiles('go.mod') }}
+        restore-keys: ${{ env.GO_CACHE_PREFIX }}
+    - name: script
+      run: |
+        go run funcgen-cd/funcgen.go --compile=true --path=./ --out-path=/tmp/funcgen/ --var-file-path=/tmp/funcgen/sources.tfvars.json --prefix=johnny-
+  deploy-github:
+    name: Build and push Github Fetcher image
+    runs-on: ubuntu-20.04
+    needs: [check-generated, terraform, test, test-cf]
+    if: github.event_name != 'pull_request' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/production')
```
Comment (lwsanty) [inline .github/workflows/push.yml:100]: will it deploy for `olek` branch?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      run: |
+        set -x
+        VERSION=${GITHUB_REF#refs/heads/}-${GITHUB_SHA:0:7}
+        export GITHUB_IMAGE_BRANCH_COMMIT=${GITHUB_DOCKER_IMAGE}:${VERSION}
+        make gcloud-push-image-github
+  deploy-jira:
+    name: Build and push Jira Fetcher image
+    runs-on: ubuntu-20.04
+    needs: [check-generated, terraform, test, test-cf]
+    if: github.event_name != 'pull_request' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/production')
```
Comment (lwsanty) [inline .github/workflows/push.yml:116]: will it deploy for `olek` branch?

**Review by lwsanty — CHANGES_REQUESTED** [review]

**Review by dennwc — APPROVED** [review]

File: .github/workflows/push.yml (hunk 1 above)
Comment (vmarkovtsev) [reply .github/workflows/push.yml:100]: No, it matches the current behavior of `.travis.yml`. Do we want it?
```yaml
stages:
  - name: test
  - name: deploy
    if: branch IN (master, production) AND type != pull_request
  - name: deploy-cf
    if: branch IN (master, production) AND type != pull_request
```

Comment (vmarkovtsev) [reply .github/workflows/push.yml:100]: Oooooh now I get it, sorry https://athenian.slack.com/archives/CQUFEUP5H/p1605864799049300?thread_ts=1605864714.048800&cid=CQUFEUP5H

**Review by lwsanty — APPROVED** [review]

**Pull request merged**